### DISCUSSION TITLE
Veramo SDR and Zero Knowledge Proof

### DISCUSSION THREAD

**Discussion by andra961** [decentralized-identity/veramo#770] [Q&A] [answered]
Hi, as you probably now, beneath the notions of Self Sovraign Identity lies the Zero Knowledge Proof. According to that, a VC holder who possess a single VC with multiple claims, would be able to only present the single claims the verifier needs, while hiding the rest of them in the process. From what i see in Veramo,  such thing is not possible trough the SDR flow followed by `createVerifiblePresentation()`, since a presentation is merely an aggregation of existing VCs, with all of their claims exposed. Am i missing something or is this the way it currently works?

**Comment by mirceanis** [comment, marked as answer]
You are not missing anything, this is the way it currently works.

The recommended pattern for the current selective disclosure is to use atomic credentials (credentials with a single claim).
This is because it is impossible to derive **verifiable** credentials using only a subset of their claims with the currently supported crypto operations.

We plans to support other crypto primitives that can make this possible (#605)(#333), but I have no estimate for that; we are a very small team.

If your project(s) depend on this feature, then please make a proposal as to how you think this API could look.

**Reply by navnita123** [reply]
Is this implemeted? I also want to derive subset of verifiable credential and create verifiable presentation of that

**Reply by mirceanis** [reply]
It is not implemented as a core Veramo plugin (deployed from this repository), but there are 3rd party implementations. Veramo is designed to work with plugins and it doesn't matter where the plugins are coming from. Feel free to experiment.

There is a tentative PR by @veromassera (#1408) at making a core plugin for BBS+ but it is stalled because of compatibility issues. It's best if that plugin is developed independently of this repository as we try to only publish multi-platform compatible plugins from here.
